tensorboard/plugins/custom_scalar/README.md

```diff
@@ -0,0 +1,175 @@
+# The Custom Scalar Dashboard
+
+## The Problem this Dashboard Solves
+
+The (other) scalar dashboard shows line charts for data collected from scalar
+summaries: Each line chart visualizes data for 1 tag, and tags are automatically
+organized into categories based on how tags are prefixed.
+
+Hence, during data collection (adding summaries to the model), some users might
+be compelled to construct tag strings in ways that codge the scalar dashboard to
+render in deseriable layouts. This is suboptimal: The display of data is coupled
+with the collection of data.
+
+For instance, a user might introduce 2 tags (say `rms/foo` and `rms/bar`) that
+each pertain to loss for different models. The scalar dashboard introduces 1
+chart for each tag, making it harder to compare loss for the 2 models.
+
+In addition, some users desire to organize the dashboard in ways that differ
+from the default. For instance, some users would like to quickly see charts for
+certain loss functions above the fold of the viewport.
+
+## Overview
+
+In turn, the *custom* scalar dashboard lets users
+
+1. Create line charts with custom combinations of runs and tags by assigning
+each chart a list of regular expressions for tags.
+2. Lay out the dashboard in a customized way.
+
+To use this dashboard, users first collect scalar data and then lay out the UI.
+
+### Collecting Scalar Data
+
+As expected, users first
+[collect scalar data with summaries](https://www.tensorflow.org/get_started/summaries_and_tensorboard).
+For example:
+
+```python
+from tensorboard import summary as summary_lib
+...
+loss = tf.losses.softmax_cross_entropy(
+    onehot_labels=tf.one_hot(labels, depth=num_classes),
+    logits=logits,
+    weights=weights)
+summary_lib.scalar('cross_entropy', loss)
+```
+
+Some code might use `tf.summary.scalar` from the *TensorFlow* library in lieu of
+`summary_lib.scalar`. That works as well and will continue to be supported by
+TensorBoard.
+
+### Specifying a Layout
+
+To lay out the dashboard, users pass a `layout_pb2.Layout` proto to the
+`summary_lib.custom_scalars_pb` method. This method does not return a TensorFlow
+op but rather a Summary proto that be passed to the SummaryWriter (See sample
+code.).
+
+The `Layout` proto (See layout.proto within this directory for docs) organizes
+the entire dashboard and contains a list of `Category`'s.
+
+Each `Category` corresponds to a collapsible section within the dashboard. Each
+`Category` has 3 fields.
+
+* **title**: A string field denoting what to label the collapsible.
```

Comment:
Just asking: is it worth duplicating the documentation already on layout.proto itself here?  It seems like it might be easier to just include the pointer, or at most give a very high level "categories are collapsible sections, containing charts which specify tag regexes" kind of description.  Otherwise we risk getting out of sync w.r.t. the actual proto file.

Reply:
That is a good point. Furthermore, we already provide example code below for creating a proto.

How about we just link to layout.proto like this?